flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -528,27 +490,6 @@ public void run(SourceContext<String> ctx) throws Exception {
         public void cancel() {}
     }
 
-    /** If finishing a task doesn't swallow exceptions this test would fail with an exception. */
-    @Test
-    public void finishingIgnoresExceptions() throws Exception {
```

Comment:
test removal explanation: I removed the `finishTask`. We also do not swallow exceptions anymore. We assume the contract of `SourceFunction#cancel` which is called when triggering a synchronous checkpoint, and expect the source to finish gracefully. If it does not the exception is not swallowed, as it is a user error.